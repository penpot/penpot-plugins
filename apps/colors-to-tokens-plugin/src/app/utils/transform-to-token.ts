import { LibraryColor } from '@penpot/plugin-types';
import { TokenStructure } from '../../model';

function transformToRgba({
  color,
  opacity,
}: Required<Pick<LibraryColor, 'color' | 'opacity'>>) {
  color = color.slice(1);

  const r = parseInt(color.substring(0, 2), 16);
  const g = parseInt(color.substring(2, 4), 16);
  const b = parseInt(color.substring(4, 6), 16);

  return `rgba(${r}, ${g}, ${b}, ${opacity})`;
}

interface Color extends LibraryColor {
  color: string;
}

export function transformToToken(colors: LibraryColor[]) {
  const result: TokenStructure = {};

  colors
    .filter((data): data is Color => !!data.color)
    .forEach((data) => {
      const currentOpacity = data.opacity ?? 1;
      const value =
        currentOpacity === 1
          ? data.color
          : transformToRgba({
              opacity: currentOpacity,
              color: data.color,
            });
      const names: string[] = data.name.split(' ');
      const key: string = data.path.replace(' \\/ ', '/').replace(/ /g, '');

      if (!result[key]) {
        result[key] = {};
      }

      const props = [key, ...names];
      let acc = result;

      props.forEach((prop, index) => {
        if (!acc[prop]) {
          acc[prop] = {};
        }

        if (index === props.length - 1) {
          let propIndex = 1;
          const initialProp = prop;

          while (acc[prop]?.$value) {
            prop = `${initialProp}${propIndex}`;
            propIndex++;
          }

          acc[prop] = {
            $value: value,
            $type: 'color',
          };
        }

        acc = acc[prop] as TokenStructure;
      });
    });

  return result;
}
