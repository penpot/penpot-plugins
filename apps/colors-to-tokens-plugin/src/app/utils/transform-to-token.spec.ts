import { expect, test } from 'vitest';
import { transformToToken } from './transform-to-token';

const initColors = [
  {
    name: 'dm chart yellow',
    color: '#ecc94b',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'dm text blue',
    color: '#63b3ed',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'green 700',
    color: '#276749',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'dm text green emphasis',
    color: '#9ae6b4',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm icon blue',
    color: '#2a4365',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm card background',
    color: '#ffffff',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'gray 600',
    color: '#4a5568',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'yellow 200',
    color: '#faf089',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm button',
    color: '#E2E8F0',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'green 300',
    color: '#68d391',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'dm chart red',
    color: '#e53e3e',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'gray 400',
    color: '#a0aec0',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm dashboard background',
    color: '#EDF2F7',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm text yellow',
    color: '#744210',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'pink 100',
    color: '#fed7e2',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm text secondary',
    color: '#718096',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'blue 900',
    color: '#1a365d',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'green 800',
    color: '#22543d',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'red 300',
    color: '#FC8181',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm button yellow',
    color: '#fefcbf',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm text red emphasis',
    color: '#e53e3e',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'dm chart green',
    color: '#38a169',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm placeholder',
    color: '#718096',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'purple 500',
    color: '#805ad5',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm chart yellow',
    color: '#faf089',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'shadow light',
    color: '#000000',
    opacity: 0.16078432,
    path: 'colors',
  },
  {
    name: 'dm footer',
    color: '#1a202c',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'red 800',
    color: '#822727',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm text blue',
    color: '#2a4365',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'purple 300',
    color: '#b794f4',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'purple 100',
    color: '#e9d8fd',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'dm background blue',
    color: '#1a365d',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'dm text red default',
    color: '#f56565',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'gray 700',
    color: '#2d3748',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm button blue',
    color: '#bee3f8',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'dm icon default',
    color: '#e2e8f0',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'green 600',
    color: '#2f855a',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'yellow 100',
    color: '#fefcbf',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'blue 100',
    color: '#bee3f8',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'dm background yellow',
    color: '#5f370e',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'blue 650',
    color: '#2a4365',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'dm text green default',
    color: '#68d391',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm bar line',
    color: '#a0aec0',
    opacity: 0.4,
    path: 'ui / light mode',
  },
  {
    name: 'dm background green',
    color: '#1c4532',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'dm button blue',
    color: '#2b6cb0',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'gray 100',
    color: '#edf2f7',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'dm button yellow',
    color: '#975a16',
    opacity: 1,
    path: 'ui / dark mode',
  },
  /* 3 different blue colors in the same path */
  {
    name: 'lm icon blue',
    color: '#bee3f8',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm icon blue',
    color: '#0000ff',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm text default',
    color: '#2d3748',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm icon red',
    color: '#c53030',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm text green default',
    color: '#22543d',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm icon green',
    color: '#276749',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'shadow dark',
    color: '#000000',
    opacity: 0.69803923,
    path: 'colors',
  },
  {
    name: 'lm chart background',
    color: '#E2E8F0',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'yellow 050',
    color: '#fffff0',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'green 100',
    color: '#c6f6d5',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm sidebar',
    color: '#1a202c',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'dm label',
    color: '#a0aec0',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'green 050',
    color: '#f0fff4',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'dm button green',
    color: '#2f855a',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm label',
    color: '#4a5568',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm button active',
    color: '#a0aec0',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'dm icon secondary',
    color: '#718096',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'dm chart background',
    color: '#4a5568',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm axis line',
    color: '#a0aec0',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'dm button yellow[DONTUSE]',
    color: '#fefcbf',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'dm sidebar',
    color: '#171923',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'dm text emphasis',
    color: '#edf2f7',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm chart green',
    color: '#68d391',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm background blue',
    color: '#ebf8ff',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'red 100',
    color: '#FED7D7',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'dm text default',
    color: '#a0aec0',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'red 500',
    color: '#e53e3e',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'yellow 800',
    color: '#744210',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'blue 050',
    color: '#ebf8ff',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'gray 800',
    color: '#1a202c',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm background yellow',
    color: '#fffff0',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'dm chart accent',
    color: '#9f7aea',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm shadow',
    color: '#000000',
    opacity: 0.2,
    path: 'ui / light mode',
  },
  {
    name: 'dm text red emphasis',
    color: '#FC8181',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'dm input',
    color: '#4a5568',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'dm chart accent alt',
    color: '#ecc94b',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm button green',
    color: '#c6f6d5',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'red 600',
    color: '#c53030',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm icon secondary',
    color: '#e2e8f0',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'dm text yellow',
    color: '#faf089',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'green 500',
    color: '#38a169',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'shadow mid',
    color: '#000000',
    opacity: 0.6,
    path: 'colors',
  },
  {
    name: 'lm text green emphasis',
    color: '#38a169',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'blue 700',
    color: '#2c5282',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm text red default',
    color: '#822727',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm footer',
    color: '#e2e8f0',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm text emphasis',
    color: '#000000',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'dm card background',
    color: '#2d3748',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm chart accent',
    color: '#b794f4',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'white',
    color: '#ffffff',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'dm button default',
    color: '#4a5568',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm input',
    color: '#EDF2F7',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm chart accent alt',
    color: '#faf089',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'gray 200',
    color: '#e2e8f0',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm icon yellow',
    color: '#744210',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'dm dashboard background',
    color: '#1a202c',
    opacity: 1,
    path: 'ui / dark mode',
  },
  {
    name: 'lm icon default',
    color: '#A0AEC0',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'yellow 700',
    color: '#975a16',
    opacity: 1,
    path: 'colors',
  },
  {
    name: 'lm background green',
    color: '#f0fff4',
    opacity: 1,
    path: 'ui / light mode',
  },
  {
    name: 'lm chart red',
    color: '#FC8181',
    opacity: 1,
    path: 'ui / light mode',
  },
];

test('transform colors to tokens', () => {
  const result = transformToToken(initColors);

  expect(result).toMatchSnapshot();
});
