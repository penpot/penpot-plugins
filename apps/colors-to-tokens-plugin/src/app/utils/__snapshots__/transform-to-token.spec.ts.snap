// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform colors to tokens 1`] = `
{
  "colors": {
    "blue": {
      "050": {
        "$type": "color",
        "$value": "#ebf8ff",
      },
      "100": {
        "$type": "color",
        "$value": "#bee3f8",
      },
      "650": {
        "$type": "color",
        "$value": "#2a4365",
      },
      "700": {
        "$type": "color",
        "$value": "#2c5282",
      },
      "900": {
        "$type": "color",
        "$value": "#1a365d",
      },
    },
    "gray": {
      "100": {
        "$type": "color",
        "$value": "#edf2f7",
      },
      "200": {
        "$type": "color",
        "$value": "#e2e8f0",
      },
      "400": {
        "$type": "color",
        "$value": "#a0aec0",
      },
      "600": {
        "$type": "color",
        "$value": "#4a5568",
      },
      "700": {
        "$type": "color",
        "$value": "#2d3748",
      },
      "800": {
        "$type": "color",
        "$value": "#1a202c",
      },
    },
    "green": {
      "050": {
        "$type": "color",
        "$value": "#f0fff4",
      },
      "100": {
        "$type": "color",
        "$value": "#c6f6d5",
      },
      "300": {
        "$type": "color",
        "$value": "#68d391",
      },
      "500": {
        "$type": "color",
        "$value": "#38a169",
      },
      "600": {
        "$type": "color",
        "$value": "#2f855a",
      },
      "700": {
        "$type": "color",
        "$value": "#276749",
      },
      "800": {
        "$type": "color",
        "$value": "#22543d",
      },
    },
    "pink": {
      "100": {
        "$type": "color",
        "$value": "#fed7e2",
      },
    },
    "purple": {
      "100": {
        "$type": "color",
        "$value": "#e9d8fd",
      },
      "300": {
        "$type": "color",
        "$value": "#b794f4",
      },
      "500": {
        "$type": "color",
        "$value": "#805ad5",
      },
    },
    "red": {
      "100": {
        "$type": "color",
        "$value": "#FED7D7",
      },
      "300": {
        "$type": "color",
        "$value": "#FC8181",
      },
      "500": {
        "$type": "color",
        "$value": "#e53e3e",
      },
      "600": {
        "$type": "color",
        "$value": "#c53030",
      },
      "800": {
        "$type": "color",
        "$value": "#822727",
      },
    },
    "shadow": {
      "dark": {
        "$type": "color",
        "$value": "#00000069",
      },
      "light": {
        "$type": "color",
        "$value": "#00000016",
      },
      "mid": {
        "$type": "color",
        "$value": "#00000060",
      },
    },
    "white": {
      "$type": "color",
      "$value": "#ffffff",
    },
    "yellow": {
      "050": {
        "$type": "color",
        "$value": "#fffff0",
      },
      "100": {
        "$type": "color",
        "$value": "#fefcbf",
      },
      "200": {
        "$type": "color",
        "$value": "#faf089",
      },
      "700": {
        "$type": "color",
        "$value": "#975a16",
      },
      "800": {
        "$type": "color",
        "$value": "#744210",
      },
    },
  },
  "ui/darkmode": {
    "dm": {
      "background": {
        "blue": {
          "$type": "color",
          "$value": "#1a365d",
        },
        "green": {
          "$type": "color",
          "$value": "#1c4532",
        },
        "yellow": {
          "$type": "color",
          "$value": "#5f370e",
        },
      },
      "button": {
        "blue": {
          "$type": "color",
          "$value": "#2b6cb0",
        },
        "default": {
          "$type": "color",
          "$value": "#4a5568",
        },
        "green": {
          "$type": "color",
          "$value": "#2f855a",
        },
        "yellow": {
          "$type": "color",
          "$value": "#975a16",
        },
        "yellow[DONTUSE]": {
          "$type": "color",
          "$value": "#fefcbf",
        },
      },
      "card": {
        "background": {
          "$type": "color",
          "$value": "#2d3748",
        },
      },
      "chart": {
        "accent": {
          "$type": "color",
          "$value": "#9f7aea",
          "alt": {
            "$type": "color",
            "$value": "#ecc94b",
          },
        },
        "background": {
          "$type": "color",
          "$value": "#4a5568",
        },
        "green": {
          "$type": "color",
          "$value": "#38a169",
        },
        "red": {
          "$type": "color",
          "$value": "#e53e3e",
        },
        "yellow": {
          "$type": "color",
          "$value": "#ecc94b",
        },
      },
      "dashboard": {
        "background": {
          "$type": "color",
          "$value": "#1a202c",
        },
      },
      "footer": {
        "$type": "color",
        "$value": "#1a202c",
      },
      "icon": {
        "default": {
          "$type": "color",
          "$value": "#e2e8f0",
        },
        "secondary": {
          "$type": "color",
          "$value": "#718096",
        },
      },
      "input": {
        "$type": "color",
        "$value": "#4a5568",
      },
      "label": {
        "$type": "color",
        "$value": "#a0aec0",
      },
      "sidebar": {
        "$type": "color",
        "$value": "#171923",
      },
      "text": {
        "blue": {
          "$type": "color",
          "$value": "#63b3ed",
        },
        "default": {
          "$type": "color",
          "$value": "#a0aec0",
        },
        "emphasis": {
          "$type": "color",
          "$value": "#edf2f7",
        },
        "green": {
          "default": {
            "$type": "color",
            "$value": "#68d391",
          },
          "emphasis": {
            "$type": "color",
            "$value": "#9ae6b4",
          },
        },
        "red": {
          "default": {
            "$type": "color",
            "$value": "#f56565",
          },
          "emphasis": {
            "$type": "color",
            "$value": "#FC8181",
          },
        },
        "yellow": {
          "$type": "color",
          "$value": "#faf089",
        },
      },
    },
  },
  "ui/lightmode": {
    "lm": {
      "axis": {
        "line": {
          "$type": "color",
          "$value": "#a0aec0",
        },
      },
      "background": {
        "blue": {
          "$type": "color",
          "$value": "#ebf8ff",
        },
        "green": {
          "$type": "color",
          "$value": "#f0fff4",
        },
        "yellow": {
          "$type": "color",
          "$value": "#fffff0",
        },
      },
      "bar": {
        "line": {
          "$type": "color",
          "$value": "#a0aec040",
        },
      },
      "button": {
        "$type": "color",
        "$value": "#E2E8F0",
        "active": {
          "$type": "color",
          "$value": "#a0aec0",
        },
        "blue": {
          "$type": "color",
          "$value": "#bee3f8",
        },
        "green": {
          "$type": "color",
          "$value": "#c6f6d5",
        },
        "yellow": {
          "$type": "color",
          "$value": "#fefcbf",
        },
      },
      "card": {
        "background": {
          "$type": "color",
          "$value": "#ffffff",
        },
      },
      "chart": {
        "accent": {
          "$type": "color",
          "$value": "#b794f4",
          "alt": {
            "$type": "color",
            "$value": "#faf089",
          },
        },
        "background": {
          "$type": "color",
          "$value": "#E2E8F0",
        },
        "green": {
          "$type": "color",
          "$value": "#68d391",
        },
        "red": {
          "$type": "color",
          "$value": "#FC8181",
        },
        "yellow": {
          "$type": "color",
          "$value": "#faf089",
        },
      },
      "dashboard": {
        "background": {
          "$type": "color",
          "$value": "#EDF2F7",
        },
      },
      "footer": {
        "$type": "color",
        "$value": "#e2e8f0",
      },
      "icon": {
        "blue": {
          "$type": "color",
          "$value": "#2a4365",
        },
        "blue1": {
          "$type": "color",
          "$value": "#bee3f8",
        },
        "blue2": {
          "$type": "color",
          "$value": "#0000ff",
        },
        "default": {
          "$type": "color",
          "$value": "#A0AEC0",
        },
        "green": {
          "$type": "color",
          "$value": "#276749",
        },
        "red": {
          "$type": "color",
          "$value": "#c53030",
        },
        "secondary": {
          "$type": "color",
          "$value": "#e2e8f0",
        },
        "yellow": {
          "$type": "color",
          "$value": "#744210",
        },
      },
      "input": {
        "$type": "color",
        "$value": "#EDF2F7",
      },
      "label": {
        "$type": "color",
        "$value": "#4a5568",
      },
      "placeholder": {
        "$type": "color",
        "$value": "#718096",
      },
      "shadow": {
        "$type": "color",
        "$value": "#00000020",
      },
      "sidebar": {
        "$type": "color",
        "$value": "#1a202c",
      },
      "text": {
        "blue": {
          "$type": "color",
          "$value": "#2a4365",
        },
        "default": {
          "$type": "color",
          "$value": "#2d3748",
        },
        "emphasis": {
          "$type": "color",
          "$value": "#000000",
        },
        "green": {
          "default": {
            "$type": "color",
            "$value": "#22543d",
          },
          "emphasis": {
            "$type": "color",
            "$value": "#38a169",
          },
        },
        "red": {
          "default": {
            "$type": "color",
            "$value": "#822727",
          },
          "emphasis": {
            "$type": "color",
            "$value": "#e53e3e",
          },
        },
        "secondary": {
          "$type": "color",
          "$value": "#718096",
        },
        "yellow": {
          "$type": "color",
          "$value": "#744210",
        },
      },
    },
  },
}
`;
